lib/Utility/Diagnostics.php

```diff
@@ -0,0 +1,223 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\Utility;
+
+
+class Diagnostics
+{
+    public function checkToken() {
+
+        global $sugar_config;
+
+        if (!isset($_REQUEST['callerToken']) || !isset($sugar_config['cache_dir'])) {
+            return false;
+        }
+
+        $tokenFilename = $sugar_config['cache_dir'] . $_REQUEST['callerToken'] . '.tmp';
+
+        $timeStamp = filemtime($tokenFilename);
+        if (($timeStamp === false) || // token file doesn't exist
+            (time() - $timeStamp > 30)) { // token file validity is expired
+            return false;
+        }
+
+        return true;
+    }
+
+    public function QueryWebServerFromCLI()
+    {
+        global $sugar_config;
+
+        if (!isset($sugar_config['cache_dir'])) {
+            return '';
+        }
+
+        //Generate a random token and save it as a filename in a short-lived file:
+        $token = bin2hex(openssl_random_pseudo_bytes(32));
+        $tokenFilename = $sugar_config['cache_dir'] . $token .'.tmp';
+        $handle = sugar_fopen($tokenFilename, "w");
+
+        // from this point on, once the token file is created, we want to ensure we always delete it in the end, no matter what
+        try {
+            fclose($handle);
+
+            $post_data = [
+                'callerToken' => $token,
+                'module' => 'Administration',
+                'action' => 'DiagnosticQuickReport',
+            ];
+            // always make the call as localhost, but get any port number from the site_url, if present
+            $url = preg_replace('/(.*):\/\/(.*?)((?::[0-9]+)|\/)(.*)/', '$1://localhost$3$4', $sugar_config['site_url']);
+            $url .= '/index.php?entryPoint=queryWebServerFromCLI';
```

Comment:
Hi @pgorod, calling an entryPoint together with an action is no longer allowed

Reply:
I actually don't need the action there, I will remove it.